pkg/network/encoding/marshal/usm_http_test.go

```diff
@@ -307,6 +310,201 @@ func TestLocalhostScenario(t *testing.T) {
+// with both pre-NAT and post-NAT connections as seen in Kubernetes environments
+func TestKubernetesLocalNATScenario(t *testing.T) {
+	if runtime.GOOS == "windows" || os.Getenv("CI") == "true" {
+		t.Skip("Skipping test on Windows or CI")
+	}
+	assert := assert.New(t)
+
+	// Create IP addresses and ports
+	clientIP := util.AddressFromString("172.29.161.37")
+	k8sServiceIP := util.AddressFromString("10.100.103.122") // K8s Service IP (ClusterIP)
+	serverIP := util.AddressFromString("172.29.191.94")
+	clientPort := uint16(53792)
+	servicePort := uint16(7778)
+	serverPort := uint16(7777)
+
+	// Create both connections exactly as seen from SUSM-94 reproduction environment
+	connections := []network.ConnectionStats{
+		{ // Pre-NAT connection: client -> k8s service (this is from the client's perspective, pid is the client pid)
+			ConnectionTuple: network.ConnectionTuple{
+				Source:    clientIP,
+				SPort:     clientPort,
+				Dest:      k8sServiceIP,
+				DPort:     servicePort,
+				Pid:       1784258,
+				Type:      network.TCP,
+				Direction: network.OUTGOING,
+			},
+			ProtocolStack: protocols.Stack{Application: protocols.HTTP},
+			// Post-NAT translation, coming from the conntrack module
+			IPTranslation: &network.IPTranslation{
+				// NOTE: this is flipped compared to what one might expect
+				// The ReplSrc fields contain the server endpoint, and ReplDst contains the client
+				ReplSrcIP:   serverIP,
+				ReplSrcPort: serverPort,
+				ReplDstIP:   clientIP,
+				ReplDstPort: clientPort,
+			},
+		},
+		{ // client -> server (from server's perspective, pid is the server pid)
+			ConnectionTuple: network.ConnectionTuple{
+				Source:    serverIP,
+				SPort:     serverPort,
+				Dest:      clientIP,
+				DPort:     clientPort,
+				Pid:       334392,
+				Type:      network.TCP,
+				Direction: network.INCOMING,
+			},
+			ProtocolStack: protocols.Stack{Application: protocols.HTTP},
+			// No IPTranslation for this connection
+		},
+	}
+
+	// Create HTTP stats with a sample request - 4 requests as seen in logs
+	httpStats := http.NewRequestStats()
+	httpStats.AddRequest(200, 15.0, 0, nil)
+	httpStats.AddRequest(200, 15.0, 0, nil)
+	httpStats.AddRequest(200, 15.0, 0, nil)
+	httpStats.AddRequest(200, 15.0, 0, nil)
+
+	// Create pre-NAT HTTP key (client → k8s service)
+	preNATKey := http.NewKey(
+		clientIP,
+		k8sServiceIP,
+		clientPort,
+		servicePort,
+		[]byte("/delay/5"),
+		true,
+		http.MethodGet,
+	)
+
+	// Create post-NAT HTTP key (client -> server)
+	// This is assuming that USM normalize the connection key, meaning that the clientIP
+	// will always be in the saddr field (this is what we've seen when reproducing SUSM-94)
+	postNATKey := http.NewKey(
+		clientIP,   // Client is still the source in the HTTP key (172.29.161.37)
+		serverIP,   // Server is the destination (172.29.191.94)
+		clientPort, // Client port (53792)
+		serverPort, // Server port (7777)
+		[]byte("/delay/5"),
+		true,
+		http.MethodGet,
+	)
```

Comment:
If the only difference between the `preNATKey` and the `postNATKey` is the `serverIP`, let's structure it more cleanly by creating a `http.newKey` and updating just that field.

Plus IMO the comments are not really relevant